### PR TITLE
add data limit fetch implementation

### PR DESCRIPTION
**Pull request by followthemoney1**
# Why are we making this pull request?
Original problem described: https://github.com/cph-cachet/flutter-plugins/pull/581

# Updates
Such as original plugins already contain `limit` parameter in ios native files, we could basically try to add this parameter to `getHealthDataFromTypes` function.

# How to use
```
 ///This is only answer does the user requested permission or not
  ///It's doesent mean that he allow to read\write
  ///[https://stackoverflow.com/a/33021234]
  Future<HealthDataPermission> requestAccessToHealthDataGranted() async {
    final bool accessWasGranted = await _health.requestAuthorization(_allTypes);

    localDataSource.saveHealthAccess(accessWasGranted ? HealthDataPermission.granted : HealthDataPermission.forbidden);

    return accessWasGranted ? HealthDataPermission.granted : HealthDataPermission.forbidden;
  }
```

Callout: https://github.com/cph-cachet/flutter-plugins/pull/706

### CODE REVIEW

File: packages/health/lib/src/health_data_point.dart [hunk 1]
```diff
@@ -12,17 +12,10 @@ class HealthDataPoint {
   String _deviceId;
   String _sourceId;
   String _sourceName;
+  String _uuid;
```
Comment (hoffmatteo) [inline packages/health/lib/src/health_data_point.dart:15]: Is there a specific reason to add this new field?

File: packages/health/lib/src/health_factory.dart [hunk 2]
```diff
@@ -426,35 +385,34 @@ class HealthFactory {
...
-      DateTime startTime, DateTime endTime, HealthDataType dataType) async {
+  Future<List<HealthDataPoint>> _dataQuery(DateTime startTime, DateTime endTime, HealthDataType dataType,
+      {int? limit}) async {
     final args = <String, dynamic>{
       'dataTypeKey': dataType.name,
       'dataUnitKey': _dataTypeToUnit[dataType]!.name,
       'startTime': startTime.millisecondsSinceEpoch,
-      'endTime': endTime.millisecondsSinceEpoch
+      'endTime': endTime.millisecondsSinceEpoch,
+      if (limit != null && limit > -1) 'limit': limit,
```
Comment (hoffmatteo) [inline packages/health/lib/src/health_factory.dart:415]: Since it is already implemented on the iOS side this is a great change!

**Review by hoffmatteo — COMMENTED** [review]
I am currently unable to test on iOS so I am not sure if I will be able to include this in the upcoming release until I am able to test, but this is a great feature.

File: packages/health/lib/src/health_data_point.dart (hunk 1 above)
Comment (followthemoney1) [reply packages/health/lib/src/health_data_point.dart:15]: @hoffmatteo added only for simplify cuz plugin is not returning an uuid in general this means that we need to use another plugins to get uuid of user. In general this is more a draft which you can change if needed

File: packages/health/lib/src/health_factory.dart (hunk 2 above)
Comment (followthemoney1) [reply packages/health/lib/src/health_factory.dart:415]: @hoffmatteo thanks mate 💖

**Comment by SlimShadyIAm** [comment]
Hi @followthemoney1, apologies for the delay in reviewing this PR.

There have been some changes on master - could you please resolve the merge conflicts?

**Comment by followthemoney1** [comment]
@SlimShadyIAm it's probably year passed already from those time. sorry not gonna do this to wait one more year. You can close this pr if you want

**Comment by SlimShadyIAm** [comment]
I'm sorry about that, we appreciate the effort regardless.

**Pull request closed without merging**